GIL acquire needed in ValueCache::trimPrefixes

slgong-fb opened the pull request:
Summary: Dubugged a segfault issue in Ondemand python tracing. Committing as a a separate diff from D37410204.

Test Plan:
- run a python test case with the following command for on-demand flow:
echo -e "PYTHON_STACK_TRACE=true" > /tmp/scott_kineto.conf && dyno gputrace --gputrace_duration 300ms --gpuconf /tmp/scott_kineto.conf

Reviewed By: chaekit

Differential Revision: D37662988

albanD reviewed (approved):
Thanks for the fix!

FYI @robieta

slgong-fb committed:
GIL acquire needed in ValueCache::trimPrefixes (#81061)

Summary:
Pull Request resolved: https://github.com/pytorch/pytorch/pull/81061

Dubugged a segfault issue in Ondemand python tracing. Committing as a a separate diff from D37410204.

Test Plan:
- run a python test case with the following command for on-demand flow:
echo -e "PYTHON_STACK_TRACE=true" > /tmp/scott_kineto.conf && dyno gputrace --gputrace_duration 300ms --gpuconf /tmp/scott_kineto.conf

Reviewed By: chaekit

Differential Revision: D37662988

fbshipit-source-id: eb2430854f2ddd80b4cf24951dcdb021d8bbd0e4
